taipy/gui/_renderers/builder.py

```diff
@@ -827,7 +827,10 @@ def set_value_and_default(
                             value = float(value)
                     if isinstance(value, (int, float)):
                         return self.__set_react_attribute(_to_camel_case(var_name), value)
-                self.set_attribute(_to_camel_case(var_name), value)
+                if isinstance(value, (datetime, date, time)):
```

Comment:
I would have wriitten this:
```suggestion
                if isinstance(value, (datetime, date, time)):
                    value = _date_to_string(value)
                self.set_attribute(_to_camel_case(var_name), value)
```
That to me is more readable.